frontend/src/pages/modelServing/screens/projects/kServeModal/EnvironmentVariablesSection.tsx

```diff
@@ -16,31 +16,44 @@ import {
 } from '@patternfly/react-icons';
 import { UpdateObjectAtPropAndValue } from '~/pages/projects/types';
 import { CreatingInferenceServiceObject } from '~/pages/modelServing/screens/types';
+import { ServingContainer } from '~/k8sTypes';
 
 type EnvironmentVariablesSectionType = {
-  data?: CreatingInferenceServiceObject;
+  data: CreatingInferenceServiceObject;
   setData: UpdateObjectAtPropAndValue<CreatingInferenceServiceObject>;
 };
 
-const EnvironmentVariablesSection: React.FC<EnvironmentVariablesSectionType> = () => {
-  const [additionalEnvVars, setAdditionalEnvVars] = React.useState<
-    Array<{ name: string; value: string }>
-  >([]);
+const EnvironmentVariablesSection: React.FC<EnvironmentVariablesSectionType> = ({
+  data,
+  setData,
+}) => {
+  const [additionalEnvVars, setAdditionalEnvVars] = React.useState<ServingContainer['env']>(
+    data.servingRuntimeEnvVars,
+  );
 
   const addEnvVar = () => {
-    setAdditionalEnvVars((prevVars) => [...prevVars, { name: '', value: '' }]);
+    if (additionalEnvVars) {
+      const newVars = [...additionalEnvVars, { name: '', value: '' }];
+      setAdditionalEnvVars(newVars);
+      setData('servingRuntimeEnvVars', newVars);
```

Comment:
We're duplicating state here - now that the `data.servingRuntimeEnvVars` exists, I don't think we still need our `additionalEnvVars` useState here. Every place that updates it also updates the data in the form state, so we can just rely on those `setData` calls and use `data.servingRuntimeEnvVars` I think.